content/articles/pathlib-vs-os-modules-for-file-and-directory-system-operation/index.md

```diff
@@ -0,0 +1,919 @@
+### Pathlib vs OS Modules for File and Directory System Operation
+### Prerequisites
+- Basic knowledge in Python programming language.
+### Contents
+- [Introduction](#introduction)
+- [Os module](#os-module)
+- [Pathlib module](#pathlib-module)
+- [Theoretical difference between Pathlib and Os](#theoretical-difference-between-Pathlib-and-OS)
+- [Practical difference between Os and Pathlib module](#practical-difference-between-OS-and-Pathlib-module)
+- [Querying current path using os and pathlib](#querying-current-path-using-os-and-pathlib)
+- [Navigating to home directory](#navigating-to-home-directory)
+- [Listing directory content](#listing-directory-content)
+- [Creating a directory and files](#creating-a-directory-and-files)
+- [Checking for an existing file or directory](#checking-for-an-existing-file-or-directory)
+- [Renaming a directory or file](#renaming-a-directory-or-file)
+- [Joining path together utilizing Os and Pathlib module](#joining-path-together-utilizing-os-and-pathlib-module)
+- [Querying path to retrieve a final file or directory name](#querying-path-to-retrieve-a-final-file-or-directory-name)
+- [Removing a directory](#removing-a-directory)
+- [Removing a file](#removing-a-file)
+- [Retrieving and converting a file information](#retrieving-and-converting-a-file-information)
+- [Querying a file extension](#querying-a-file-extension)
+- [Wrapping Up](#wrapping-up)
+
+### Introduction
+File System Operation can be characterized as the example or technique that can be
+used to control or construct files and directories and there are two different ways
+of working with File System Operation, either utilizing functional dependency or utilizing
+Object-Oriented. I will practically highlight the different situations to separate between
+the two kinds of modules python provided for system operation. These modules are OS and Pathlib.
+
+#### OS Module
+The OS (operating system) module is a different operating system interface that functionally renders a convenient way
+of working with files and directories with a simple function call, additionally,
+OS empowers us to deal with I/O tasks straightforwardly.
+#### Pathlib Module
+The pathlib module is an Object-Oriented document framework activity that gives an order of just
+one direct superclass "PurePath" which permits you to play out a computational activity yet, doesn't
+give I/O activities, yet a substantial classpath that broadens PurePath and gives procedure on I/O.
+
+### Theoretical difference between Pathlib and Os
```

Comment:
You can move this section at the last. The reader must first gain an insight about the concept before trying to understand the difference

Reply:
Okay